### ISSUE TITLE
<Page/> - Page.Content breaks on safari when nested divs have margin bottom or top

### ISSUE THREAD

**Issue by tzachbon** [wix/wix-style-react#5824]
# 🐛 Bug Report

### 🏗 Relevant Components

`Page.Content`

### 😯 Current Behavior

On `safari` when we used nested divs with margin bottom or top it broke the UI with react error:
```
Invariant Violation
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
```

### 👣 Steps to Reproduce

We are not sure why it happens but after a lot of checking we came to conclusion that it related to the [height calculation in the page content](https://github.com/wix/wix-style-react/blob/81f810b67f2d274f027531be7ea6593703861754/src/Page/Page.js#L117) and the related safari bug with `getBoundingClientRect`.
Some info about it [here](https://muffinman.io/ios-safari-get-bounding-client-rect-bug/).

You can see it in the [playground](https://www.wix.com/pages/wix-style-react/?path=/story/introduction-playground--playground&snippet=f70b1a9c-6ec0-41a0-bafe-19bf8a3f3280) open in `safari`

### 👀 Severity

-  Major

**Comment by tyv** [comment]
I have the same in chrome
![image](https://user-images.githubusercontent.com/122360/90419687-9b791300-e0bf-11ea-82f5-628ba52593f6.png)

the content without page wrapper works well.

**Comment by tyv** [comment]
simple example is not easy to derive, but I can give you branch for experiments: 
`npm start` this package: https://github.com/wix-private/new-releases/tree/tyv/wsr-page-bug/wix-new-releases-backoffice-ui

you will see the list, click on any item.
this code will be rendered: https://github.com/wix-private/new-releases/blob/tyv/wsr-page-bug/wix-new-releases-backoffice-ui/src/components/AutoViewsApp/AutoViewsApp.tsx#L95-L108

**Comment by shlomitc** [comment]
Thanks, guys.
We'll try tackling this in the upcoming weeks.
@tyv - if you manage to create a reproducible example in the playground, that would be great
https://www.wix-style-react.com/?path=/story/introduction-playground--playground

**Comment by tyv** [comment]
@shlomitc it will require installing external libs

**Comment by EgoziE** [comment]
I managed to reproduce it on Chrome as well, the problem is really deep.

This code fails:
```
<Page>
  <Page.Content>
    <div style={{ margin: '10px' }}>xxx</div>
  </Page.Content>
</Page>
```

In order to solve it quickly wrap the content with the wanted margin as padding like so:
```
<Page>
  <Page.Content>
    <div style={{ padding: '10px' }}>
      <div>xxx</div>
    </div>
  </Page.Content>
</Page>
```

From what I understand the problem is related to margin collapsing that we abuse unintentionally when trying to force a min-height on the content container.
A possible fix can be changing the container display from block to flex.

**Comment by EgoziE** [comment]
fixed in 9.20.0
